src/SFA.DAS.ApprenticeCommitments/Infrastructure/SqlServerConnectionFactory.cs

```diff
@@ -1,14 +1,29 @@
 ﻿using System.Data.Common;
+using Microsoft.Azure.Services.AppAuthentication;
 using Microsoft.Data.SqlClient;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using SFA.DAS.ApprenticeCommitments.Extensions;
 
 namespace SFA.DAS.ApprenticeCommitments.Infrastructure
 {
     public class SqlServerConnectionFactory : IConnectionFactory
     {
+        private readonly IConfiguration _configuration;
+
+        public SqlServerConnectionFactory(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
         public DbContextOptionsBuilder<TContext> AddConnection<TContext>(DbContextOptionsBuilder<TContext> builder, string connection) where TContext : DbContext
         {
-            return builder.UseSqlServer(connection);
+            var sqlConnection = new SqlConnection(connection)
+            {
+                AccessToken = GetAccessToken(),
+            };
```

Comment:
Might it be (very slightly!) more explicit to simply call `CreateConnection` rather than describe how the connection is created?

Reply:
yes good point